src/time_evolution/mesolve.jl

```diff
@@ -98,71 +103,61 @@ where
 - `prob::ODEProblem`: The ODEProblem for the master equation time evolution.
 """
 function mesolveProblem(
-    H::QuantumObject{MT1,HOpType},
-    ψ0::QuantumObject{<:AbstractArray{T2},StateOpType},
+    H::Union{AbstractQuantumObject{DT1,HOpType},Tuple},
+    ψ0::QuantumObject{DT2,StateOpType},
     tlist,
     c_ops::Union{Nothing,AbstractVector,Tuple} = nothing;
     alg::OrdinaryDiffEqAlgorithm = Tsit5(),
```

Comment:
We also don't need `alg` for generating `mesolveProblem` right?

Reply:
maybe also need to check for `sesolve`, `mcsolve`, and `ssesolve`.

Reply:
Yes, we can remove it.